### PR TITLE
Unit Testing Implemented (Using jest framework)

### PR DESCRIPTION
**Pull request by NicholasMatias**
Features:
- Unit testing set up for situation, task, action, result, relevance, and frequency functions/ models
- Created a unit_testing_constants.cjs file 
- 24 test total 

Notes: 
- Had add some copies of functions into unit testing constants file to make them compatible with jest => jest uses require (common js) and is not compatible with es6 (can't use import statements). 

SS:
Screenshot 2024-07-25 at 4 46 25 PM

### CODE REVIEW

File: backend/models/unit_tests/Relevance.test.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+test("getRelevanceScore(question, matching response)", () => {
+    const relevanceScore = getRelevanceScore(
+        strengthQuestion,
+        strengthQuestionResponse
+    )[1];
+    expect(relevanceScore).toBeGreaterThan(0.7);
+});
+
+test("getRelevanceScore(question, matching response)", () => {
```
Comment (ocorrodi) [inline backend/models/unit_tests/Relevance.test.js:16]: nit: would be great to differentiate test names by expected result (like question, unrelated response)

File: backend/models/unit_tests/Frequency.test.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    expect(freq[1][1]).toBe(2); // expected = 2
+    expect(freq[2][1]).toBe(1); // see = 1
+});
+
+test("getFreq(text, n = 3, ngramSize = 2) => 3 most common 2 word phrases", () => {
+    const freq = getFreq(testFreq, 3, 2);
+    expect(freq[0][1]).toBe(2); // as expected = 2
+    expect(freq[1][1]).toBe(1); // this is = 1
+    expect(freq[2][1]).toBe(1); // is a = 1
+});
```
Comment (ocorrodi) [inline backend/models/unit_tests/Frequency.test.js:17]: For future thought/unit test dev: can sometimes be good to think of edge cases, like what if ngramSize = 0 or if ngramSize > len(text) etc. Even though we don't necessarily expect these sorts of inputs normally it's good to know that it still works as intended under weird scenarios.

**Review by ocorrodi — COMMENTED** [review]
nice job! unit testing can be tedious but is a good skill to have and an important part of development :)
